source/util/hex_float.h

```diff
@@ -889,12 +1030,86 @@ ParseNormalFloat<FloatProxy<Float16>, HexFloatTraits<FloatProxy<Float16>>>(
 
   // Overflow on 16-bit behaves the same as for 32- and 64-bit: set the
   // fail bit and set the lowest or highest value.
+  // /!\ We get an error if there is no overflow but the value is infinity.
```

Comment:
I think the idea I had was that if you wanted an infinity to be explicitly represented, you should use hex floats to do it.


See also line 863 where it says:

```
  if (val.isInfinity()) {
    // Fail the parse.  Emulate standard behaviour by setting the value to
    // the closest normal value, and set the fail bit on the stream.
    value.set_value((value.isNegative() | negate_value) ? T::lowest()
                                                        : T::max());
    is.setstate(std::ios_base::failbit);
  }
```

(I regret very much calling these methods ParseNormalFloat.  Really it's parsing decimal floats.)